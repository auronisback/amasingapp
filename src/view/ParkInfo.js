import React, {Component} from 'react';

export default class ParkInfo extends Component {
  /**
   * Renderizes the element.
   * */
  render() {
    return (
        <div className={'ParkInfoWindow container'}>
          <div className={'row'}>Occupancy Rate: {(this.props.occRate * 100).toFixed(0)}%</div>
          <div className={'row'}>Distance: {ParkInfo._distanceToString(this.props.distance)}</div>
          <div className={'row'}>Park Number: {this.props.parkNumber}</div>
          <div className={'row'}>Type: {this.props.streetType}</div>
          <div className={'row'}>Had measure: {this.props.haveMes ? 'Yes' : 'No'}</div>
        </div>
    );
  }

  /**
   * Converts a distance in a pretty printable form.
   * @param[in] distance The distance from the destination
   * @return string The string representing distance
   * */
  static _distanceToString(distance) {
    let d = (+distance).toFixed(0);
    if(d < 1000) // Short distance, returning it in meters
      return d + 'm';
    else
      return (d / 1000).toFixed(1) + 'km';
  };
}