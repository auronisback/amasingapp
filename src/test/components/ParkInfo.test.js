import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import ParkInfo from "../../components/ParkInfo";

Enzyme.configure({adapter: new Adapter()});

describe('<ParkInfo>', () => {
  it('Should render without crashing', () => {
    const parkInfo = shallow(<ParkInfo
        occRate={0.12}
        distance={321}
        parkNumber={6}
        streetType={"residential"}
        haveMes={true}
    />);
    expect(parkInfo).toMatchSnapshot();
  });
});