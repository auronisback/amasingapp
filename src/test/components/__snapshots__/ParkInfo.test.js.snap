// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ParkInfo> Should render without crashing 1`] = `
<div
  className="ParkInfoWindow container"
>
  <div
    className="row"
  >
    Occupancy Rate: 
    12
    %
  </div>
  <div
    className="row"
  >
    Distance: 
    321m
  </div>
  <div
    className="row"
  >
    Park Number: 
    6
  </div>
  <div
    className="row"
  >
    Type: 
    residential
  </div>
  <div
    className="row"
  >
    Had measure: 
    Yes
  </div>
</div>
`;
